src/pyhf/tensor/pytorch_backend.py

```diff
@@ -219,6 +224,7 @@ def simple_broadcast(self, *args):
             list of Tensors: The sequence broadcast together.
         """
 
+        args = [arg if arg.size() != torch.Size([]) else arg.view(1) for arg in args]
```

Comment:
There may be a more elegant way to avoid the 0-d tensor TyperErrors

```python
>>> import pyhf
>>> import torch
>>> torch.Size(pyhf.tensorlib.astensor(1.))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/feickert/.venvs/pyhf-dev/lib/python3.7/site-packages/torch/tensor.py", line 456, in __iter__
    raise TypeError('iteration over a 0-d tensor')
TypeError: iteration over a 0-d tensor
```

Reply:
do we ever broadcast on scalars? 


`[arg.view(1) if not self.shape(arg) else arg for arg in args]`  ?

Reply:
we could also think about adding a `atleast_1d` method 

https://www.tensorflow.org/api_docs/python/tf/experimental/numpy/atleast_1d
https://pytorch.org/docs/master/generated/torch.atleast_1d.html
https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.atleast_1d.html

Reply:
> do we ever broadcast on scalars?

We don't in our internals, but you could imagine a situation in which someones ends up accessing the backend tools and does.

> we could also think about adding a atleast_1d method

Yeah, this seems the best.

Reply:
https://pytorch.org/docs/master/generated/torch.atleast_1d.html is from `master` and so looks like is targeting `torch` `v1.7.0` (so at least 2 releases away from now). We should probably make this an Issue and then move froward with a solution for now.

Reply:
@lukasheinrich I like your

```python
[arg.view(1) if not self.shape(arg) else arg for arg in args]
```

pattern, so I'll use that here. :+1: